packages/datadog-plugin-aws-sdk/src/services/kinesis.js

```diff
@@ -11,6 +11,58 @@ class Kinesis {
       'aws.kinesis.stream_name': params.StreamName
     })
   }
+
+  // AWS-SDK will b64 kinesis payloads
+  // or will accept an already b64 encoded payload
+  // This method handles both
+  static tryParse (body, final = false) {
+    try {
+      return JSON.parse(body)
+    } catch (e) {
+      if (final) {
+        return null
+      }
+      return Kinesis.tryParse(Buffer.from(body, 'base64').toString('ascii'), true)
+    }
+  }
+
+  requestInject (span, request, tracer) {
+    const operation = request.operation
+    if (operation === 'putRecord' || operation === 'putRecords') {
+      try {
+        if (!request.params) {
```

Comment:
This doesn't need to be in the try/catch block.

Reply:
Actually, it looks like a lot of the below code won't throw and shouldn't be in the try/catch block.